.github/scripts/build_assets/filehandler.py

```diff
@@ -0,0 +1,143 @@
+import json
+from zipfile import ZipFile
+from pathlib import Path
+from typing import List
+import os
+import re
+
+
+def find_new_icons(devicon_json_path: str, icomoon_json_path: str) -> List[dict]:
+    """
+    Find the newly added icons by finding the difference between
+    the devicon_test.json and the icomoon_test.json.
+    :param devicon_json_path, the path to the devicon_test.json.
+    :param icomoon_json_path: a path to the iconmoon.json.
+    :return: a list of the new icons as JSON objects.
+    """
+    with open(devicon_json_path) as json_file:
+        devicon_json = json.load(json_file)
+
+    with open(icomoon_json_path) as json_file:
+        icomoon_json = json.load(json_file)
+
+    new_icons = []
+    for icon in devicon_json:
+        if is_not_in_icomoon_json(icon, icomoon_json):
+            new_icons.append(icon)
+
+    return new_icons
+
+
+def is_not_in_icomoon_json(icon, icomoon_json) -> bool:
+    """
+    Checks whether the icon's name is not in the icomoon_json.
+    :param icon: the icon object we are searching for.
+    :param icomoon_json: the icomoon json object parsed from
+    icomoon_test.json.
+    :return: True if icon's name is not in the icomoon_test.json, else False.
+    """
+    pattern = re.compile(f"^{icon['name']}-")
+
+    for font in icomoon_json["icons"]:
+        if pattern.search(font["properties"]["name"]):
+            return False
+    return True
+
+
+def get_svgs_paths(new_icons: List[dict], icons_folder_path: str) -> List[str]:
+    """
+    Get all the suitable svgs file path listed in the devicon_test.json.
+    :param new_icons, a list containing the info on the new icons.
+    :param icons_folder_path, the path where the function can find the
+    listed folders.
+    :return: a list of svg file paths that can be uploaded to Icomoon.
+    """
+    file_paths = []
+    for icon_info in new_icons:
+        folder_path = Path(icons_folder_path, icon_info['name'])
+
+        if not (folder_path.exists() and folder_path.is_dir()):
+            print(f"Invalid path. This is not a directory: {folder_path}\nSkipping entry...")
```

Comment:
I did this due to my set up environment at the time. I was building the script in a separate repo with about 3-4 test fonts. Since I used the `icomoon.json` and the `devicon.json` from the `devicon` repo, some of the paths would not exist. I can definitely change it so that an error is thrown instead.